VMEncryption/main/check_util.py

```diff
@@ -360,7 +360,24 @@ def validate_volume_type_for_enable(self, public_settings, existing_volume_type)
             return
         raise Exception('Moving from volume type {0} to volume type {1} is not allowed'.format(existing_volume_type, volume_type))
 
-    def precheck_for_fatal_failures(self, public_settings, encryption_status, DistroPatcher, existing_volume_type):
+    def validate_skr_release_rsa_or_ec_key(self,public_settings,check_release_rsa_or_ec_key):
+        if not check_release_rsa_or_ec_key:
+            return
+        imds_res_id = public_settings.get(CommonVariables.EncryptionManagedIdentity)
+        KeyEncryptionKeyUrl=public_settings.get(CommonVariables.KeyEncryptionKeyURLKey)
+        if imds_res_id:
+            os.environ["IMDS_MSI_RES_ID"]=imds_res_id
+        cmd = './AzureAttestSKR -n 123456 -k {0} -c imds -r'.format(KeyEncryptionKeyUrl)
+        executor = CommandExecutor(self.logger)
+        result = executor.Execute(cmd)
+        if result != CommonVariables.process_success:
+            msg='Identity used in VM can not do wrapKey/unwrapKey operations.'
```

Comment:
Suggestion: "The managed identity provided in public settings is not correct or not authorized to do secure key release operation in your key vault/HSM. Please verify and re-run ADE install after fixing the issue"